### PR TITLE
Add support for multiple ignored directives

### PR DESCRIPTION
**Pull request by skaldo**
Sometimes it is useful to have a possibility to ignore more than one directive.

### CODE REVIEW

**Comment by Narretz** [comment]
Hi, I have a few questions. What exactly does this change and why do we need it? Also, tests are failing and the change itself must be tested.

**Comment by skaldo** [comment]
Hi, this change allows us to pass an array containing the names of the directives, we don't want to compile for some reason, to the $compile function.
Regarding tests, I've run them on my fork and the last two have passed, the 1st and 2nd failed because of timeouts..

**Comment by Narretz** [comment]
Travis says there are some failures in the e2e tests. They don't look like they are timeouts / flakes. Also, we still need a test for your change.

File: src/ng/compile.js [hunk 1]
```diff
@@ -2104,6 +2104,11 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
     function addDirective(tDirectives, name, location, maxPriority, ignoreDirective, startAttrName,
                           endAttrName) {
       if (name === ignoreDirective) return null;
+      if (angular.isArray(ignoreDirective)) {
+        for (var i = 0; i < ignoreDirective.length; i++) {
+          if (name === ignoreDirective[i]) return null;
+        }
+      }
```
Comment (gkalpak) [inline src/ng/compile.js]: This whole block can be reduced to:

```js
if (isArray(ignoreDirective) && includes(ignoreDirective, name)) return null;
```

**Comment by gkalpak** [comment]
I am a bit sceptical, because this is undocumented (private) API, but this change will not be serving the needs of Angular core.

**Comment by skaldo** [comment]
Well, in our application this change is really beneficial. On the other hand, if it's going to be useless for the core and if it's big job to make it working, let's close this request.

Anyway, what's failing the tests is the recursive call of compileNodes function with the ignoreDirective parameter.

**Comment by caitp** [comment]
you know, this probably could be useful for internal purposes, like helping people build recursive tree structured directives easily using the directive API. It would need a bit more than that, but I can see how you could build that kind of API on top of it.

**Comment by gkalpak** [comment]
As pointed out by @Narretz, this is essentially the same functionality as #14635. Although this also has test coverage, #14635's approach is more powerful/flexible, so I'm going to close this in favor of #14635.
<sub>(I doubt we are likely to implement this feature, but if we would #14635 would be the way to go.)</sub>

**Pull request closed without merging**